### PR TITLE
[VL] Daily Update Velox Version (2024_06_30)

### PR DESCRIPTION
**Pull request by GlutenPerfBot**
Upstream Velox's New Commits:

```txt
0ef0ac8e4 by Jia Ke, Enable right join in smj (10148)
c54e59dbb by wypb, Fix HashStringAllocator::clear() and cumulativeBytes_ (10260)
4963d7116 by duanmeng, Add recursive spill for RowNumber (8654)
e3de4ea9d by Sandino Flores, Add support for Protobuf v22+ (10294)
0d8022846 by PHILO-HE, Support finding installed arrow libraries from system (9992)
fd955bff4 by liangyongyuan, Add float/double types support for Spark mod function (9848)
0ced9e5f0 by NEUpanning, Fix typo in expression evaluation documentation (10304)
8803bfbd1 by lingbin, Fix typo in SIMD document (10319)
bcfc8f8c3 by PHILO-HE, Allow returning Status from callNullable and callNullFree methods (10274)
258db516d by PHILO-HE, Use legacySizeOfNull argument to determine the behavior of Spark size function (10100)
```

### CODE REVIEW

**Comment by PHILO-HE** [comment]
@marin-ma, we still need to apply some changes for arrow build (if arrow lib is not found, velox will resolve the dependency by 
 building from source as before). Its cmake file is moved here:
https://github.com/facebookincubator/velox/tree/main/CMake/resolve_dependency_modules/arrow

**Comment by PHILO-HE** [comment]
ARROW_HOME needs to be set to  

"${VELOX_HOME}/_build/release/CMake/resolve_dependency_modules/arrow/arrow_ep/" (replace release with debug for debug mode).

https://github.com/apache/incubator-gluten/blob/main/cpp/CMakeLists.txt#L71-L73

**Review by marin-ma — APPROVED** [review]

**Comment by FelixYBW** [comment]
@PHILO-HE just to confirm, with this PR, how many times Arrow will be built and where it's built?

File: backends-velox/src/test/scala/org/apache/gluten/execution/TestOperator.scala [hunk 1]
```diff
@@ -1017,7 +1017,7 @@ class TestOperator extends VeloxWholeStageTransformerSuite with AdaptiveSparkPla
     }
   }
 
-  test("test explode/posexplode function") {
```
Comment (FelixYBW) [inline backends-velox/src/test/scala/org/apache/gluten/execution/TestOperator.scala:1020]: @marin-ma why these operators test should be ignored?

**Comment by PHILO-HE** [comment]
> @PHILO-HE just to confirm, with this PR, how many times Arrow will be built and where it's built?

@FelixYBW, currently, there are still two builds, since a build issue for Velox was found yesterday. We need wait for this velox pr: https://github.com/facebookincubator/velox/pull/10355

**Comment by FelixYBW** [comment]
Can we add this PR to OAP/Velox temporarily? 

In future if a PR from us to Velox may impact Gluten, let's submit the PR to Gluten also and make sure once Velox's PR is merged Gluten can work as expected.

**Pull request merged**